crates/context_aware_config/src/api/config/handlers.rs

```diff
@@ -656,3 +661,33 @@ async fn get_resolved_config(
 
     Ok(resp.json(response))
 }
+
+#[get("/versions")]
+async fn get_config_versions(
+    db_conn: DbConnection,
+    filters: Query<QueryFilters>,
+) -> superposition::Result<Json<PaginatedResponse<ConfigVersion>>> {
+    let DbConnection(mut conn) = db_conn;
+
+    let n_version: i64 = config_versions::config_versions
+        .count()
+        .get_result(&mut conn)?;
+    let mut builder = config_versions::config_versions
+        .into_boxed()
+        .order(config_versions::created_at.desc());
+    if let Some(limit) = filters.count {
+        builder = builder.limit(limit);
+    }
+    if let Some(page) = filters.page {
+        let offset = (page - 1) * filters.count.unwrap_or(10);
```

Comment:
@mahatoankitkumar can we handle the negative value for offset as it is not allowed in offset and giving 5xx
`OFFSET must not be negative`